lib/puppetserver/ca/action/generate.rb

```diff
@@ -133,7 +191,15 @@ def generate_certs(certnames, settings, digest)
           ca = Puppetserver::Ca::CertificateAuthority.new(@logger, settings)
 
           passed = certnames.map do |certname|
-            key, csr = generate_key_csr(certname, settings, digest)
+            if !alt_names.empty?
+              # When validating the cert, OpenSSL will ignore the CN field if
+              # altnames are present, so we need to ensure that the certname is
+              # also listed among the alt names.
+              alt_names += ",DNS:#{certname}"
```

Comment:
I would assume that each cert in the certnames list would have the `DNS:#{certname}` value of the preceding certs? It'd be nice to set something like
```rb
this_certs_alt_names = alt_names.dup
if !this_certs_alt_names.empty?
  this_certs_alt_names += ",DNS:#{certname}"
```

Reply:
Oh, derp